libobs/util/platform-nix.c

```diff
@@ -1033,3 +1034,32 @@ uint64_t os_get_free_disk_space(const char *dir)
 
 	return (uint64_t)info.f_frsize * (uint64_t)info.f_bavail;
 }
+
+#if !defined(__APPLE__)
+void os_random_bytes(void *buffer, uint32_t length)
+{
+	getrandom(buffer, length, 0);
```

Comment:
`getrandom` should also be called in a loop like `fread` as its not guaranteed to read `length` bytes in one call unless `length` < 256

Reply:
Good point. It's unlikely any consumers will need to generate such a large amount of randomness but it should be handled properly. I'll get this fixed.